pilot/cmd/pilot-agent/main.go

```diff
@@ -267,8 +266,7 @@ var (
 			secOpts.EnableGatewaySDS = enableGatewaySDSEnv
 			secOpts.CAProviderName = caProviderEnv
 
-			// TODO: extract from ProxyConfig
-			secOpts.TrustDomain = trustDomainEnv
+			secOpts.TrustDomain = trustDomain
```

Comment:
Why ? This is the flag which we're trying to deprecate. Please use the trustDomain from ProxyConfig - or the env variable.